src/main/java/it/gov/pagopa/pu/pagopapayments/service/synchronouspayments/SynchronousPaymentService.java

```diff
@@ -75,6 +81,35 @@ private InstallmentDTO getPayableDebtPositionByOrganizationAndNav(Organization o
   }
 
   public long retrieveNotificationFeeCents(Long organizationId, String nav, String accessToken){
+    DebtPositionTypeOrg debtPositionTypeOrg = debtPositionService.findDebtPositionTypeOrgByOrgIdAndNavAndOrigins(organizationId, nav, ORDINARY_DEBT_POSITION_ORIGINS, accessToken);
+    if(debtPositionTypeOrg!=null && Boolean.TRUE.equals(debtPositionTypeOrg.getFlagAmountActualization())) {
+        return retrieveNotificationFeeCentsFromPuSil(debtPositionTypeOrg, nav, accessToken);
+    } else {
+        return retrieveNotificationFeeCentsFromSend(organizationId, nav, accessToken);
+    }
+  }
+
+  private long retrieveNotificationFeeCentsFromPuSil(DebtPositionTypeOrg debtPositionTypeOrg, String nav, String accessToken) {
+    try{
+      if(debtPositionTypeOrg.getAmountActualizationOrgSilServiceId()!=null)
+      {
+        log.info("Retrieve notification fee from pu-sil by OrgSilServiceId {} and nav {}", debtPositionTypeOrg.getAmountActualizationOrgSilServiceId(), nav);
+        AmountUpdatesDTO amountUpdatesDTO = puSilService.getAmountUpdates(debtPositionTypeOrg.getAmountActualizationOrgSilServiceId(), nav, accessToken);
+        if (amountUpdatesDTO.getNotificationFee()!=null && amountUpdatesDTO.getNotificationFee()>0)
+          return amountUpdatesDTO.getNotificationFee();
+      }else {
+        log.error("Failed to retrieve notification fee from pu-sil because OrgSilServiceId is null");
```

Comment:
```suggestion
        log.error("Failed to retrieve notification fee from pu-sil because amountActualizationOrgSilServiceId is null on debtPositionTypeOrgId {}", debtPositionTypeOrg.getDebtPositionTypeOrgId());
```